src/ManipulationMsg.chpl

```diff
@@ -58,87 +58,84 @@ module ManipulationMsg {
-            return ret;
-          }
+        // use List;
+        // var bcDimsList = new list(int);
+        // for i in 0..<ndIn do if bcDims[i] then bcDimsList.pushBack(i);
+
+        // // iterate over each slice of the output array corresponding to one
+        // // copy of the input array and perform the copy
+        // /* Example:
+        //   broadcast 5x1x3 array into 5x4x3 array:
+        //   - the 5x1x3 array is copied into the 5x4x3 array 4 times
+        //   - domOffAxis => {0..0, 0..<4, 0..0}
+        //   - for 'nonBCIndex' = (0, 0, 0), outSliceIdx = (0..<5, 0..0, 0..<3)
+        //   - for 'nonBCIndex' = (0, 1, 0), outSliceIdx = (0..<5, 1..1, 0..<3)
+        //   - etc.
+        // */
+        // forall nonBCIndex in domOffAxis(eOut.a.domain, bcDimsList.toArray()) {
+        //   const nbcT = if ndOut == 1 then (nonBCIndex,) else nonBCIndex;
+
+
+        //   var outSliceIdx: ndOut*range;
+        //   for i in 0..<ndOut do outSliceIdx[i] = 0..<shapeOut[i];
+        //   for i in 0..<ndIn do if bcDims[i] then outSliceIdx[i] = nbcT[i];
+
+        //   eOut.a[(...outSliceIdx)] = eIn.a; // !!! Doesn't work because of rank mismatch !!!
+        // }
 
-          // copy values from the input array into the output array
-          forall idx in eOut.a.domain with (var agg = newSrcAggregator(t), in bcDims) {
-            const idxIn = imap(if ndOut==1 then (idx,) else idx, bcDims);
-            agg.copy(eOut.a[idx], eIn.a[idxIn]);
-          }
+        inline proc imap(idx: array_nd_out*int, bc: array_nd_in*int): array_nd_in*int {
+          var ret: array_nd_in*int;
+          for param i in 0..<array_nd_in do ret[i] = if bc[i] then 0 else idx[i + (array_nd_out - array_nd_in)];
+          return ret;
         }
-      }
-
-      return st.insert(eOut);
-    }
 
-    select gEnt.dtype {
-      when DType.Int64 do return doBroadcast(int);
-      when DType.UInt8 do return doBroadcast(uint(8));
-      when DType.UInt64 do return doBroadcast(uint);
-      when DType.Float64 do return doBroadcast(real);
-      when DType.Bool do return doBroadcast(bool);
-      otherwise {
-        var errorMsg = notImplementedError(getRoutineName(),gEnt.dtype);
-        mLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-        return MsgTuple.error(errorMsg);
+        // copy values from the input array into the output array
+        forall idx in eOut.a.domain with (var agg = newSrcAggregator(array_dtype), in bcDims) {
+          const idxIn = imap(if array_nd_out==1 then (idx,) else idx, bcDims);
+          agg.copy(eOut.a[idx], eIn.a[idxIn]);
+        }
       }
     }
+
+    return st.insert(eOut);
+  }
+
+  proc broadcastToMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab,
+    type array_dtype,
+    param array_nd_in: int,
+    param array_nd_out: int
+  ): MsgTuple throws
+    where array_nd_in > array_nd_out
+  {
```

Comment:
can version of `broadcastToMsg` also get called even though it doesn't have the `@arkouda.instantiateAndRegister`?

Reply:
yeah, the generated code will just call `ManipulationMsg.broadcastToMsg(...)`, and then it's up to the compiler to decide which overload to call from there.

As a side note, I'd like to add support for where-clause interpretation to `instantiateAndRegister`, s.t. it wouldn't even try to instantiate a function whenever the where-clause would evaluate to false. In this case, that would make this second overload unnecessary.